### PR TITLE
doc: add the upgrade guide from 5.0 to 2022.1

### PR DESCRIPTION
**Pull request by annastuchlik**
Fix https://github.com/scylladb/scylla-docs/issues/4125

I've added the upgrade guides from 5.0 to 2022.1. They are based on the previous upgrade guides from Open Source to Enterprise.

### CODE REVIEW

**Comment by tzach** [comment]
@roydahan please review
Any exceptions to 5.0 to 2022.1 upgrade we should include?

File: docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.0-to-2022.1/metric-update-5.0-to-2022.1.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
+=================================================================
+ScyllaDB Metric Update - Scylla 5.0 to Scylla Enterprise 2022.1
+=================================================================
+
+New Metrics
+------------
```
Comment (tzach) [inline docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.0-to-2022.1/metric-update-5.0-to-2022.1.rst:6]: AFAIK, there are **no** metric updates from 5.0 to 2022.1
@amnonh please advise

Comment (amnonh) [reply docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.0-to-2022.1/metric-update-5.0-to-2022.1.rst:6]: I couldn't find any updates

**Comment by roydahan** [comment]
> @roydahan please review
> Any exceptions to 5.0 to 2022.1 upgrade we should include?

There is one that we’re checking now and base on the decision in the issue we may need to change.
https://github.com/scylladb/scylla-enterprise-machine-image/issues/50#issuecomment-1195146977

File: docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.0-to-2022.1/metric-update-5.0-to-2022.1.rst (hunk 1 above)
Comment (annastuchlik) [reply docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.0-to-2022.1/metric-update-5.0-to-2022.1.rst]: @tzach I've updated the page about metrics.

File: docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.0-to-2022.1/upgrade-guide-from-5.0-to-2022.1-rpm.rst [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+To upgrade:
+
+#. Update the `ScyllaDB RPM Enterprise repo <http://www.scylladb.com/enterprise-download/centos_rpm/>`_  to **2022.1**.
+#. Install:
+
+    .. code:: sh
+
+       sudo yum clean all
+       sudo rm -rf /var/cache/yum
+       sudo yum remove scylla\*
```
Comment (roydahan) [inline docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.0-to-2022.1/upgrade-guide-from-5.0-to-2022.1-rpm.rst:89]: @annastuchlik / @tzach in case the base version (5.0 or earlier) was installed using a scylla image (e.g. AMI).
There is another step that should be taken here:

For  Ubuntu20:
'''
sudo apt-get install scylla-enterprise-machine-image -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold"
'''

For CentOS (old AMI that was upgraded to 5.0):
'''
sudo yum install scylla-enterprise-machine-image
'''

Comment (roydahan) [reply docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.0-to-2022.1/upgrade-guide-from-5.0-to-2022.1-rpm.rst:89]: @soyacz please review as well

Comment (soyacz) [reply docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.0-to-2022.1/upgrade-guide-from-5.0-to-2022.1-rpm.rst:89]: I'm not sure if we need to add `-o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold"` - this setting makes apt not to update modified config files (without prompt). But anyway we restore config files from backup later.
We use this part in our tests because this is the best way to answer prompts automatically.

Comment (roydahan) [reply docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.0-to-2022.1/upgrade-guide-from-5.0-to-2022.1-rpm.rst:89]: Edited my comment.

Comment (annastuchlik) [reply docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.0-to-2022.1/upgrade-guide-from-5.0-to-2022.1-rpm.rst:89]: @roydahan I've updated the installation step. Could you re-review?

File: docs/upgrade/_common/upgrade-guide-from-5.0-to-2022.1-ubuntu-and-debian.rst [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+       sudo apt-get clean all
+       sudo apt-get update
+       sudo apt-get remove scylla\*
+       sudo apt-get install scylla-enterprise
+       for conf in $(cat /var/lib/dpkg/info/scylla-*server.conffiles /var/lib/dpkg/info/scylla-*conf.conffiles /var/lib/dpkg/info/scylla-*jmx.conffiles | grep -v init ) /etc/systemd/system/{var-lib-scylla,var-lib-systemd-coredump}.mount; do sudo cp -v $conf.backup-5.0 $conf; done
+       sudo systemctl daemon-reload
+
+    Answer ‘y’ to the first two questions.
+
+    If you installed version 5.0 using a ScyllaDB image (for example, AMI) or first installed an earlier version of ScyllaDB
```
Comment (soyacz) [inline docs/upgrade/_common/upgrade-guide-from-5.0-to-2022.1-ubuntu-and-debian.rst:100]: I'd rather rephrase it to:
'If you use cloud image with preinstalled version of Scylla, install additional package  `scylla-enterprise-machine-image` with command `sudo apt-get install scylla-enterprise-machine-image`'

this is for all debian/ubuntu guides.

For guides related to centos, command is different: `sudo yumt install scylla-enterprise-machine-image` (yum instead of apt-get).

**Review by soyacz — CHANGES_REQUESTED** [review]

File: docs/upgrade/_common/upgrade-guide-from-5.0-to-2022.1-ubuntu-and-debian.rst (hunk 3 above)
Comment (annastuchlik) [reply docs/upgrade/_common/upgrade-guide-from-5.0-to-2022.1-ubuntu-and-debian.rst]: @soyacz Thanks, I've rephrased the paragraph.
The information CentOS is in this file: docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.0-to-2022.1/upgrade-guide-from-5.0-to-2022.1-rpm.rst
The information for Ubuntu and Debian is in the common file: docs/upgrade/_common/upgrade-guide-from-5.0-to-2022.1-ubuntu-and-debian.rst.
The file is included in all the pages for specific Ubuntu and Debian versions.

**Comment by annastuchlik** [comment]
@soyacz @roydahan I've addressed all your comments. Can you review this PR again?

**Comment by tzach** [comment]
@roydahan @soyacz please review

File: docs/upgrade/_common/upgrade-guide-from-5.0-to-2022.1-ubuntu-and-debian.rst [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+       sudo apt-get clean all
+       sudo apt-get update
+       sudo apt-get remove scylla\*
+       sudo apt-get install scylla-enterprise
+       for conf in $(cat /var/lib/dpkg/info/scylla-*server.conffiles /var/lib/dpkg/info/scylla-*conf.conffiles /var/lib/dpkg/info/scylla-*jmx.conffiles | grep -v init ) /etc/systemd/system/{var-lib-scylla,var-lib-systemd-coredump}.mount; do sudo cp -v $conf.backup-5.0 $conf; done
+       sudo systemctl daemon-reload
+
+    Answer ‘y’ to the first two questions.
+
+    If you use a cloud image with a preinstalled version of ScyllaDB (for example, AMI), you need to install an additional
```
Comment (soyacz) [inline docs/upgrade/_common/upgrade-guide-from-5.0-to-2022.1-ubuntu-and-debian.rst:100]: I think we still call `Scylla` not `ScyllaDB` aren't we?

File: docs/upgrade/_common/upgrade-guide-from-5.0-to-2022.1-ubuntu-and-debian.rst [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+
+    If you use a cloud image with a preinstalled version of ScyllaDB (for example, AMI), you need to install an additional 
+    package ``scylla-enterprise-machine-image`` with the ``sudo apt-get install scylla-enterprise-machine-image`` command:
+
+    .. code:: sh
+
+       sudo apt-get clean all
+       sudo apt-get update
+       sudo apt-get remove scylla\*
+       sudo apt-get install scylla-enterprise-machine-image
```
Comment (soyacz) [inline docs/upgrade/_common/upgrade-guide-from-5.0-to-2022.1-ubuntu-and-debian.rst:108]: better to not list it here, as users will forget about above note and install it anyway (when is not needed).

File: docs/upgrade/_common/upgrade-guide-from-5.0-to-2022.1-ubuntu-and-debian.rst (hunk 4 above)
Comment (soyacz) [reply docs/upgrade/_common/upgrade-guide-from-5.0-to-2022.1-ubuntu-and-debian.rst:100]: Also I'd move this note (and command) just before `sudo systemctl daemon-reload`

File: docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.0-to-2022.1/upgrade-guide-from-5.0-to-2022.1-rpm.rst [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+       sudo yum install scylla-enterprise 
+       for conf in $( rpm -qc $(rpm -qa | grep scylla) | grep -v contains ) /etc/systemd/system/{var-lib-scylla,var-lib-systemd-coredump}.mount; do sudo cp -v $conf.backup-5.0 $conf; done
+
+   CentOS only:
+   
+   If you use a cloud image with a preinstalled version of ScyllaDB (for example, AMI), you need to install an additional
```
Comment (soyacz) [inline docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.0-to-2022.1/upgrade-guide-from-5.0-to-2022.1-rpm.rst:95]: Here also I'd suggest moving this note after `for conf in $( rpm -qc $(rpm -qa | grep scylla) | grep -v contains ) /etc/systemd/system/{var-lib-scylla,var-lib-systemd-coredump}.mount; do sudo cp -v $conf.backup-5.0 $conf; done`

File: docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.0-to-2022.1/upgrade-guide-from-5.0-to-2022.1-rpm.rst [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+   
+   If you use a cloud image with a preinstalled version of ScyllaDB (for example, AMI), you need to install an additional 
+   package ``scylla-enterprise-machine-image`` with the ``sudo yum install scylla-enterprise-machine-image`` command:
+
+    .. code:: sh
+
+       sudo yum clean all
+       sudo rm -rf /var/cache/yum
+       sudo yum remove scylla\*
+       sudo yum install scylla-enterprise-machine-image
```
Comment (soyacz) [inline docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.0-to-2022.1/upgrade-guide-from-5.0-to-2022.1-rpm.rst:103]: the same issue like in ubuntu/debian comment.

**Review by soyacz — CHANGES_REQUESTED** [review]

**Comment by annastuchlik** [comment]
@soyacz I've updated the note according to your suggestions. 
As to Scylla vs ScyllaDB, we should be using ScyllaDB for both the company name and the database.

**Review by soyacz — APPROVED** [review]
LGTM

**Review by tzach — APPROVED** [review]
LGTM

**Review by roydahan — APPROVED** [review]

**Comment by annastuchlik** [comment]
@scylladb/scylla-maint Please backport to 5.1

**Comment by denesb** [comment]
> @scylladb/scylla-maint Please backport to 5.1

Doesn't apply cleanly to 5.1.
